crates/sui-move-build/src/linters/redundant_custom_call.rs

```diff
@@ -0,0 +1,267 @@
+    locals: BTreeMap<Var, LocalState<Value>>,
+}
+
+//**************************************************************************************************
+// impls
+//**************************************************************************************************
+
+impl SimpleAbsIntConstructor for RedundantCustomVerifier {
+    type AI<'a> = RedundantCustomVerifierAI;
+
+    fn new<'a>(
+        _env: &CompilationEnv,
+        _program: &'a Program,
+        context: &'a CFGContext<'a>,
+        _init_state: &mut <Self::AI<'a> as SimpleAbsInt>::State,
+    ) -> Option<Self::AI<'a>> {
+        let Some(_) = &context.module else {
+            return None
+        };
+        let MemberName::Function(fn_name) = context.member else {
+            return None;
+        };
+
+        Some(RedundantCustomVerifierAI {
+            fn_name_loc: fn_name.loc,
+        })
+    }
+}
+
+impl SimpleAbsInt for RedundantCustomVerifierAI {
+    type State = State;
+    type ExecutionContext = ExecutionContext;
+
+    fn finish(&mut self, _final_states: BTreeMap<Label, State>, diags: Diagnostics) -> Diagnostics {
+        diags
+    }
+
+    fn start_command(&self, _: &mut State) -> ExecutionContext {
+        ExecutionContext {
+            diags: Diagnostics::new(),
+        }
+    }
+
+    fn finish_command(&self, context: ExecutionContext, _state: &mut State) -> Diagnostics {
+        let ExecutionContext { diags } = context;
+        diags
+    }
+
+    fn exp_custom(
+        &self,
+        _context: &mut ExecutionContext,
+        _state: &mut State,
+        _e: &Exp,
+    ) -> Option<Vec<Value>> {
+        None
+    }
+
+    fn call_custom(
+        &self,
+        context: &mut ExecutionContext,
+        _state: &mut State,
+        _loc: &Loc,
+        return_ty: &Type,
+        f: &ModuleCall,
+        args: Vec<Value>,
+    ) -> Option<Vec<Value>> {
+        if let Some((_, _, fname)) = PRIVATE_OBJ_FUNCTIONS
+            .iter()
+            .find(|(addr, module, fun)| f.is(addr, module, fun))
+        {
+            if let Value::LocalObjWithStore(obj_addr_loc) = args[0] {
+                let msg = format!(
+                    "Potential unintended implementation of a custom {} function.",
+                    fname
+                );
+                let (op, action) = if *fname == TRANSFER_FUN {
+                    ("transfer", "transferred")
+                } else if *fname == SHARE_FUN {
+                    ("share", "shared")
+                } else {
+                    ("freeze", "frozen")
+                };
+                let uid_msg = format!("Instances of a type with a store ability can be {} using the \
+                                       public_{} function which often negates the intent of enforcing \
+                                       custom {} policy (implemented through calling the private {} \
+                                       function variant in the module defining this type)", action, fname, op, fname);
```

Comment:
I'm a big fan of inline usage of variables in string formatting where possible. So thoughts on this? It's a personal preference, so feel free to not apply though if you prefer the current way.

```suggestion
                let uid_msg = format!("Instances of a type with a store ability can be {action} using the \
                                       public_{fname} function which often negates the intent of enforcing \
                                       a custom {op} policy (implemented through calling the private {fname} \
                                       function variant in the module defining this type)");
```

Reply:
I don't have a strong preference so changed it accordingly